src/pfs/pfs_test.go

```diff
@@ -10,6 +10,18 @@ import (
 	"google.golang.org/protobuf/testing/protocmp"
 )
 
+func TestCommit_NilBranch(t *testing.T) {
+	var b1 = &Branch{Name: "dummy"}
+	var c1 = &Commit{Branch: b1}
+	c1.NilBranch()
+	require.Nil(t, c1.Branch)
```

Comment:
A better test would `require.NoDiff` with an expected branch.  This way you can be sure that NilBranch() doesn't also delete the Name field.